pkg/controllers/user/alert/deployer/upgradeimpl.go

```diff
@@ -140,17 +154,29 @@ func (l *alertService) Upgrade(currentVersion string) (string, error) {
 			return "", fmt.Errorf("get cluster %s failed, %v", l.clusterName, err)
 		}
 		if !v3.ClusterConditionReady.IsTrue(cluster) {
-			return "", fmt.Errorf("upgrade system service %s failed, cluster %v not ready", serviceName, l.clusterName)
+			return "", fmt.Errorf("cluster %v not ready", l.clusterName)
 		}
 
+		systemCatalog, err := l.catalogLister.Get(metav1.NamespaceAll, systemCatalogName)
+		if err != nil {
+			return "", fmt.Errorf("get catalog %s failed, %v", systemCatalogName, err)
+		}
+
+		if !v3.CatalogConditionUpgraded.IsTrue(systemCatalog) || !v3.CatalogConditionRefreshed.IsTrue(systemCatalog) || !v3.CatalogConditionDiskCached.IsTrue(systemCatalog) {
+			return "", fmt.Errorf("catalog %v not ready", systemCatalogName)
+		}
+
+		// wait for catalog syncing to latest version
+		time.Sleep(waitCatalogSyncInterval)
```

Comment:
want to understand, what's this for? we already checked catalog has been refreshed in Line 165?

Reply:
updated, also watch catalog, because latest catalog may not download completely the first time this controller execute after server upgraded